File: 1791-find-center-of-star-graph/1791-find-center-of-star-graph.cpp
class Solution {
public:
    int findCenter(vector<vector<int>>& edges) {
        unordered_map<int, int> degree;

        for (auto edge : edges) {
            degree[edge[0]]++;
            degree[edge[1]]++;
        }

        for (auto nodes : degree) {
            int node = nodes.first;
            int nodeDegree = nodes.second;

            if (nodeDegree == edges.size()) {
                return node;
            }
        }

        return -1;
    }
};

File: 1791-find-center-of-star-graph/1791-find-center-of-star-graph.java
class Solution {

    public int findCenter(int[][] edges) {
        // Since the center of the star will be in the first two edges, just check the common node
        if (edges[0][0] == edges[1][0] || edges[0][0] == edges[1][1]) {
            return edges[0][0];
        } else {
            return edges[0][1];
        }
    }
}